internal/controllers/generic/controller.go

```diff
@@ -126,6 +128,31 @@ func (c *Controller[
 	return c.reconcileNormal(ctx, adapter)
 }
 
+// shouldReconcile filters events when the object status is up to date, and its
+// status indicates that no further reconciliation is required.
+//
+// Specifically it looks at the Progressing condition. It has the following behaviour:
+// - Progressing condition is not present -> reconcile
+// - Progressing condition is present and True -> reconcile
+// - Progressing condition is present and False, but observedGeneration is old -> reconcile
+// - Progressing condition is false and observedGeneration is up to date -> do not reconcile
+//
+// If shouldReconcile is preventing an object from being reconciled which should
+// be reconciled, consider if that object's actuator is correcting returning a
```

Comment:
a typo, for real this time.
```suggestion
// be reconciled, consider if that object's actuator is correctly returning a
```